README.future.md

```diff
@@ -2,99 +2,563 @@
+
+```typescript
+const pinecone = new Pinecone();
+await pinecone.createIndex({
+  name: 'sample-index',
+  dimension: 1536,
+});
 ```
 
-It also can find information specified in the environment variables `PINECONE_API_KEY` and `PINECONE_ENVIRONMENT`.
+#### Create index with other optional configurations
 
-## Control Plane
+Many optional configuration fields allow greater control over hardware resources and availability. To learn more
+about the purpose of these fields, see [Understanding indexes](https://docs.pinecone.io/docs/indexes)
+and [Scaling indexes](https://docs.pinecone.io/docs/scaling-indexes).
+
+```typescript
+await pinecone.createIndex({
+  name: 'sample-index-2',
+  dimension: 1536,
+  metric: 'dotproduct',
+  pods: 2,
+  replicas: 2,
+  shards: 2,
+  podType: 'p1.x2',
+  metadataConfig: {
+    indexed: ['product_type'],
+  },
+});
+```
 
-### Create an index
+#### Create Index: Checking the status of a newly created index
 
-The minimum required configuration to create an index is the index name and dimension. The dimension you choose should match the output of the model used to produce your embeddings.
+The `createIndex` method issues a create request to the API that returns quickly, but the resulting index is
+not immediately ready for use upserting, querying, or performing other data operations. You can use the
+`describeIndex` method to find out the status of an index and see whether it is ready for use.
 
+```typescript
+> await pinecone.describeIndex('sample-index')
+{
+  database: {
+    name: 'sample-index',
+    dimension: 1536,
+    metric: 'cosine',
+    pods: 1,
+    replicas: 1,
+    shards: 1,
+    podType: 'p1.x1'
+  },
+  status: { ready: false, state: 'Initializing' }
+}
+> await pinecone.describeIndex('sample-index')
+{
+  database: {
+    name: 'sample-index',
+    dimension: 1536,
+    metric: 'cosine',
+    pods: 1,
+    replicas: 1,
+    shards: 1,
+    podType: 'p1.x1'
+  },
+  status: {
+    ready: true,
+    state: 'Ready',
+    host: 'sample-index-c01b9b5.svc.us-east1-gcp.pinecone.io',
+    port: 433
+  }
+}
 ```
-await client.createIndex({ name: 'my-index', dimension: 128 })
+
+#### Create Index: Waiting until the index is ready
+
+If you pass the `waitUntilReady` option, the client will handle polling for status updates on a newly created index. The promise returned by `createIndex` will not be resolved until the index status indicates it is ready to handle data operations. This can be especially useful for integration testing where index creation in a setup step will be immediately followed by data operations.
```

Comment:
```suggestion
If you pass the `waitUntilReady` option, the client will handle polling for status updates on a newly created index. The promise returned by `createIndex` will not be resolved until the index status indicates it is ready to handle data operations. This can be especially useful for integration testing, where index creation in a setup step will be immediately followed by data operations.
```